source/code-snippets/crud/arrayFilters.js

```diff
@@ -18,18 +21,28 @@ async function printData() {
 
 async function runFirstArrayElement() {
   try {
+    
+    // Get the database and collection on which to run the operation
     const myDB = client.db("test");
     const myColl = myDB.collection("testColl");
 
+    // Print the result
     console.log(JSON.stringify(await myColl.find().toArray()));
 
     // start firstArrayElement example
+    // Query for all elements in entries array where x is a string
     const query = { "entries.x": { $type : "string" } };
+
+    // For each matched element, increase value of y by 33
```

Comment:
I: I think this example only updates the first matching array element

```suggestion
    // For the first matched array entry, increase value of y by 33
```